### DISCUSSION TITLE
Relation Manager Lifecycle Hook get ownerRecord (afterAttach)

### DISCUSSION THREAD

**Discussion by moleculech** [filamentphp/filament#3488] [Help] [answered]
How can I get the ownerRecord in a Lifecycle Hook? I want to use the ownerRecord after a new record has been attached.
```php
     return $table
          ->headerActions([
              AttachAction::make()
                  ->after(function($record) {
                      $ownerRecord = ?;
                  }),
              ])
```

It was possible in the deprecated BelongsToManyRelationManager with the method afterAttach.

Thanks!

**Comment by danharrin** [comment, marked as answer]
It's `$livewire->ownerRecord`

**Reply by moleculech** [reply]
Ah, yes that works! Thanks.

                AttachAction::make()
                    ->after(function(Livewire $livewire){
                        dd($livewire->ownerRecord);
                    })
                    ->color('primary'),

 It is documented for the _Form Builder_ here: https://filamentphp.com/docs/2.x/forms/advanced
 For the _Admin Panel/Table Builder_ it is not yet documented (as far as I can tell)

Another solution would be to do it in the pivot model:

                class EventTeacher extends Pivot {
                    use HasFactory;
                
                    protected static function booted() {
                        static::creating(function($pivotModel) {
                            $teacherId = $pivotModel->teacher_id;
                            $ownerRecord = $pivotModel->pivotParent;
                    [..]

An tutorial can be found here: https://graffino.com/til/p8Ng3XPcD9-how-to-trigger-model-events-on-pivot-tables-in-laravel
